### ISSUE TITLE
 using variable for table and schema names.

### ISSUE THREAD

**Issue by rodrigodevelms** [JetBrains/Exposed#917]
Hello all. 
I'm using exposed as follows:

```
object PersonTable : Table("schema.table") {
    val id = uuid("id")
    val active = bool("active")
    val name = varchar("name", 120)
    val email = varchar("email", 120).uniqueIndex()
    val document = varchar("document", 30).uniqueIndex()
    val birthdate = date("birthdate")
    override val primaryKey: PrimaryKey = PrimaryKey(id, name = "client_id_uindex")
}
```

and

```
class ClientService : IRepository<Person> {
    override suspend fun insert(entity: Person): Any? = newSuspendedTransaction(Dispatchers.IO) {
        try {
            PersonTable.insert {
                it[id] = entity.id
                it[active] = entity.active
                it[name] = entity.name
                it[email] = entity.email
                it[document] = entity.document
                it[birthdate] = entity.birthdate
            }
            PhoneTable.batchInsert(entity.phones) { phone ->
                this[PhoneTable.id] = phone.id
                this[PhoneTable.active] = phone.active
                this[PhoneTable.number] = phone.number
                this[PhoneTable.clientFk] = phone.clientFk!!
            }
        } catch (ex: ExposedSQLException) {
            ex.printStackTrace()
            throw ex
        }
    }

```
And it works perfectly
But I need the schema and table name to be passed through parameters, and I tried it this way:

```
fun personTable(schema: String, table: String): Table {
    return object : Table("$schema.$table") {
        val id = uuid("id")
        val active = bool("active")
        val name = varchar("name", 120)
        val email = varchar("email", 120).uniqueIndex()
        val document = varchar("document", 30).uniqueIndex()
        val birthdate = date("birthdate")
        override val primaryKey: PrimaryKey = PrimaryKey(id, name = "client_id_uindex")
    }
}
```
and

```
class ClientService : IRepository<Person> {
    override suspend fun insert(entity: Person, schema: String, table:String): Any? = newSuspendedTransaction(Dispatchers.IO) {
        try {
            personTable(schema, table).insert {
                it[id] = entity.id
                it[active] = entity.active
                it[name] = entity.name
                it[email] = entity.email
                it[document] = entity.document
                it[birthdate] = entity.birthdate
            }
            PhoneTable.batchInsert(entity.phones) { phone ->
                this[PhoneTable.id] = phone.id
                this[PhoneTable.active] = phone.active
                this[PhoneTable.number] = phone.number
                this[PhoneTable.clientFk] = phone.clientFk!!
            }
        } catch (ex: ExposedSQLException) {
            ex.printStackTrace()
            throw ex
        }
    }
```

But I have errors when I try to assign the values ​​(it [value] = entity.value)
Could someone help me, I don't know how to solve the problem.

**Comment by Tapac** [comment]
You should replace it with:
```
abstract PersonBaseTable(tableName) : Table(tableName) {
    val id = uuid("id")
    val active = bool("active")
    ...
}

fun personTable(schema: String, table: String): PersonBaseTable = object : PersonBaseTable("$schema.$table")
```

**Comment by rodrigodevelms** [comment]
Thank you Tapac.

**Comment by Tapac** [comment]
@rodrigodevelms , `eagerLoading` param added to a text column to cover such cases.
Will be available in 0.24.2
